tidy3d/components/lumped_element.py

```diff
@@ -272,28 +378,504 @@ def _sheet_conductance(self):
+
+
+class RLCNetwork(Tidy3dBaseModel):
+    """Class for representing a simple network consisting of a resistor, capacitor, and inductor.
+    Provides additional functionality for representing the network as an equivalent medium.
+
+    Notes
+    -----
+
+        Implementation is based on the equivalent medium introduced by _`[1]`.
+
+        **References**
+
+        .. [1]  J. A. Pereda, F. Alimenti, P. Mezzanotte, L. Roselli and R. Sorrentino, "A new algorithm
+                for the incorporation of arbitrary linear lumped networks into FDTD simulators," IEEE
+                Trans. Microw. Theory Tech., vol. 47, no. 6, pp. 943-949, Jun. 1999.
+    """
+
+    resistance: Optional[pd.PositiveFloat] = pd.Field(
+        None,
+        title="Resistance",
+        description="Resistance value in ohms.",
+        unit=OHM,
+    )
+
+    capacitance: Optional[pd.PositiveFloat] = pd.Field(
+        None,
+        title="Capacitance",
+        description="Capacitance value in farads.",
+        unit=FARAD,
+    )
+
+    inductance: Optional[pd.PositiveFloat] = pd.Field(
+        None,
+        title="Inductance",
+        description="Inductance value in henrys.",
+        unit=HENRY,
+    )
+
+    network_topology: Literal["series", "parallel"] = pd.Field(
+        "series",
+        title="Network Topology",
+        description="Describes whether network elements are connected in ``series`` or ``parallel``.",
+    )
+
+    @cached_property
+    def _number_network_elements(self) -> pd.PositiveInt:
+        num_elements = 0
+        if self.resistance:
+            num_elements += 1
+        if self.capacitance:
+            num_elements += 1
+        if self.inductance:
+            num_elements += 1
+        return num_elements
+
+    @cached_property
+    def _to_admittance_function(self) -> tuple[tuple[float, ...], tuple[float, ...]]:
+        """Converts the :class:`RLCNetwork` instance into a rational expression representing the
+        admittance of the network in the Laplace domain.
+        """
+        R = self.resistance
+        C = self.capacitance
+        L = self.inductance
+        if self._number_network_elements == 1:
+            # Lumped element is simply a resistor, inductor, or capacitor.
+            if R:
+                return ((1, 0), (R, 0))
+            if C:
+                return ((0, C), (1, 0))
+            if L:
+                return ((1, 0), (0, L))
+        if self.network_topology == "series":
+            return RLCNetwork._to_series_network_transfer_function(R, L, C)
+        return RLCNetwork._to_parallel_network_transfer_function(R, L, C)
+
+    @staticmethod
+    def _to_series_network_transfer_function(
+        R: float, L: float, C: float
+    ) -> tuple[tuple[float, ...], tuple[float, ...]]:
+        """Converts the RLC series network to a rational expression representing the
+        admittance of the network in the Laplace domain.
+        """
+        if R and C and L:
```

Comment:
I wonder if we could be a bit more general and unified here if we define RLCNetwork recursively, as either a single R, L, or C element, or as a parallel or series connection of RLCNetworks.

Reply:
Starting think about how I might implement this, which I think is definitely possible but not exactly trivial. Basically, I would just need to apply rules for adding admittances in series or parallel, which would require performing operations on polynomials, using `numpy` or maybe `sympy` if I want to do it symbolically. I used 'lcapy' to help me with this in the notebook I linked, but I could do something similar in Tidy3D.

If you agree, I could make this functionality into a separate issue to be implemented at a future time. The reason being this PR enables the simulation of any linear network, so this would be more of an interface improvement.

Reply:
I agree this is a low priority. You can make an issue if you want. Thanks!